core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -1147,6 +1150,37 @@ public String stripPrefixIfPresent(String path) {
     T apply() throws IOException;
   }
 
+  /**
+   * Filter exception that need to be retried.
+   * if exception need to be retried will return to continue the retry.
+   * else will throw exception to quit retry.
+   * @param e exception to be handled
+   * @throws IOException Exceptions that do not need to be tried again will be thrown directly
+   */
+  private void handleRetryablException(IOException e) throws IOException {
+    try {
+      throw e;
+    } catch (AlluxioRuntimeException | AlluxioStatusException alluxioException) {
```

Comment:
i don't think any of these exception types will be caught here.. e is an IOException right?

Reply:
I think what we are doing currently is wrapping the underlying exception into IOException which is not good. If we want to handle the retryable exception, we would like to catch the specific exception in UFS, transform it into AlluxioRuntimeException and handle it gracefully here for retry.